modules/core/src/main/java/io/fluo/core/log/TracingTransaction.java

```diff
@@ -0,0 +1,164 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fluo.core.log;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import io.fluo.api.client.Snapshot;
+import io.fluo.api.client.Transaction;
+import io.fluo.api.config.FluoConfiguration;
+import io.fluo.api.config.ScannerConfiguration;
+import io.fluo.api.data.Bytes;
+import io.fluo.api.data.Column;
+import io.fluo.api.exceptions.AlreadySetException;
+import io.fluo.api.exceptions.CommitException;
+import io.fluo.api.iterator.RowIterator;
+import io.fluo.core.impl.TransactionImpl;
+import io.fluo.core.impl.TxStats;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TracingTransaction implements Transaction, Snapshot {
+
+  private static final Logger log = LoggerFactory.getLogger(FluoConfiguration.TRANSACTION_PREFIX);
+  private static final Logger collisionLog = LoggerFactory.getLogger(FluoConfiguration.TRANSACTION_PREFIX+".collisions");
+  private static final Logger summaryLog = LoggerFactory.getLogger(FluoConfiguration.TRANSACTION_PREFIX+".summary");
+  
+  private final TransactionImpl tx;
+  private final long txid;
+  private Bytes triggerRow;
+  private Column triggerColumn;
+  private Class<?> clazz;
+  private boolean committed = false;
+  
+  public TracingTransaction(TransactionImpl tx){
+    this(tx, null, null, null);
+  }
+  
+  public TracingTransaction(TransactionImpl tx, Class<?> clazz){
+    this(tx, null, null, clazz);
+  }
+  
+  public TracingTransaction(TransactionImpl tx, Bytes triggerRow, Column triggerColumn, Class<?> clazz) {
+    this.tx = tx;
+    this.txid = tx.getStartTs();
+    
+    this.triggerRow = triggerRow;
+    this.triggerColumn = triggerColumn;
+    this.clazz = clazz;
+    
+    log.trace("txid: {} begin() thread: {}", txid, Thread.currentThread().getId());
+
+    if(triggerRow != null){
+      log.trace("txid: {} trigger: {} {}", txid, triggerRow, triggerColumn);
+    }
+    
+    if(clazz != null){
+      log.trace("txid: {} class: {}", txid, clazz.getName());
+    }
+    
+  }
+
+  @Override
+  public Bytes get(Bytes row, Column column) {
+    Bytes ret = tx.get(row, column);
+    log.trace("txid: {} get({}, {}) -> {}", txid, row, column, ret);
+    return ret;
+  }
+
+  @Override
+  public Map<Column,Bytes> get(Bytes row, Set<Column> columns) {
+    Map<Column,Bytes> ret = tx.get(row, columns);
+    log.trace("txid: {} get({}, {}) -> {}", txid, row, columns, ret);
+    return ret;
+  }
+
+  @Override
+  public Map<Bytes,Map<Column,Bytes>> get(Collection<Bytes> rows, Set<Column> columns) {
+    Map<Bytes,Map<Column,Bytes>> ret = tx.get(rows, columns);
+    log.trace("txid: {} get({}, {}) -> {}", txid, rows, columns, ret);
+    return ret;
+  }
+
+  @Override
+  public RowIterator get(ScannerConfiguration config) {
+    // TODO log something better
```

Comment:
I don't mind having TODOs in the code but I think they should always correspond with an open GitHub issue or else we will never get to it:
```
// TODO  Log something better (see fluo-###)
```